packages/playwright-core/src/server/launchApp.ts

```diff
@@ -36,19 +36,21 @@ export async function launchApp(browserType: BrowserType, options: {
 }) {
   const args = [...options.persistentContextOptions?.args ?? []];
 
+  let channel = options.persistentContextOptions?.channel;
   if (browserType.name() === 'chromium') {
     args.push(
         '--app=data:text/html,',
         `--window-size=${options.windowSize.width},${options.windowSize.height}`,
         ...(options.windowPosition ? [`--window-position=${options.windowPosition.x},${options.windowPosition.y}`] : []),
         '--test-type=',
     );
+    channel ??= !options.persistentContextOptions?.executablePath ? findChromiumChannel(options.sdkLanguage) : undefined;
```

Comment:
```suggestion
    if (!channel && !options.persistentContextOptions?.executablePath)
      channel = findChromiumChannel(options.sdkLanguage);
```

Reply:
 This won't reset channel to undefined if executablePath is provided.